infra/modules/providers/azure/service-principal/variables.tf

```
@@ -7,21 +17,11 @@ variable "create_for_rbac" {
 variable "display_name" {
   description = "Display name of the AD application"
   type        = string
+  default     = "AD Client"
```

```suggestion
  default     = ""
```

I would leave this blank because it's what is expected if creating a role assignment using an existing SP. Also if blank when creating a new AAD App then a name will be generated.

---

Updated.

---

+1. Other than that this LGTM